pyproject.toml

```diff
@@ -350,6 +350,18 @@ towncrier = ">=18.6.0rc1"
 # Used for checking the Poetry lockfile
 tomli = ">=1.2.3"
 
+
+# Dependencies for building the development documentation
+[tool.poetry.group.dev-docs]
+optional = true
+
+[tool.poetry.group.dev-docs.dependencies]
+sphinx = {version = "^6.1", python = "^3.8"}
+sphinx-autodoc2 = {version = "^0.4.2", python = "^3.8"}
+myst-parser = {version = "^1.0.0", python = "^3.8"}
+furo = "^2022.12.7"
+
+
```

Comment:
Note that support for dependency groups was added in poetry core [1.1.0](https://github.com/python-poetry/poetry-core/releases/tag/1.1.0a6).

Poetry-core 1.0.8 rejects this syntax:

```
RuntimeError: The Poetry configuration is invalid:
  - Additional properties are not allowed ('group' was unexpected
```

which unfortunately means that the poetry-core range given in https://github.com/matrix-org/synapse/blob/edf046ece7c14cd8f01a4e5ebcbeefac9e0b6717/pyproject.toml#L365-L372

can no longer be built at its lower end of the range.

Options:
- Move these into the `tool.poetry.dev-dependencies` section above, or
- Bump the lower bound to >=1.1.0 and liaise with packagers, or
- ??? something else?

Actions:
- We should probably ensure something in CI tries to build a wheel using the oldest supported version of poetry-core

Reply:
It looks like systest uses poetry-core 1.1.0 (at least in offline mode) https://github.com/matrix-org/sytest/blob/cdac5a1cd09251c27dfb4666ccd6d34a223fe335/docker/synapse.Dockerfile#L50-L54

Otherwise we use the version of poetry-core bundled by poetry (the frontend) 1.3.2.